tests/validator/validator.cpp

```diff
@@ -2843,3 +2843,172 @@ TEST(Validator, duplicateIdAll)
     validator->validateModel(model);
     EXPECT_EQ_ISSUES(expectedIssues, validator);
 }
+
+TEST(Validator, circularImportReferencesComponent)
+{
+    const std::string errorMessage =
+        "Cyclic dependencies were found when attempting to resolve a component in the model 'circularImport1'. The dependency loop is:\n"
+        " - component 'i_am_cyclic' is imported from 'c2' in 'circularImport_2.cellml';\n"
+        " - component 'c2' is imported from 'c3' in 'circularImport_3.cellml';\n"
+        " - component 'c3' is imported from 'i_am_cyclic' in 'circularImport_1.cellml'; and\n"
+        " - component 'i_am_cyclic' is imported from 'c2' in 'circularImport_2.cellml'.";
```

Comment:
It's sufficient to say that we have `i_am_cyclic` -> `c2` -> `c3` -> `i_am_cyclic`. No need to say that we have `i_am_cyclic` -> `c2` -> `c3` -> `i_am_cyclic` -> `c2`.

IOW, I would remove the second `component 'i_am_cyclic' is imported from 'c2' in 'circularImport_2.cellml'`.

Reply:
3565c3f